java/fury-core/src/main/java/org/apache/fury/serializer/ArraySerializers.java

```diff
@@ -262,6 +272,13 @@ public void write(MemoryBuffer buffer, boolean[] value) {
       }
     }
 
+    @Override
+    public boolean[] copy(boolean[] originArray) {
+      boolean[] newArray = new boolean[originArray.length];
+      System.arraycopy(originArray, 0, newArray, 0, originArray.length);
```

Comment:
Could we use `Arrays.copyOf` instead for primitive types? Some JDK JIT compiler has special optimization for `Arrays.copyOf`

Reply:
No problem, I will use `Arrays.copy` to implement the copy method of primitive types.

Also, can you provide articles or links about special optimizations? I didn't find any relevant descriptions. thanks

Reply:
I forgot the original articles. You can run a jmh benchmark, and `Arrays.copy` is 2X faster than `System.arraycopy`.  
This is what we do in MemoryBuffer:
```java
  public byte[] getBytes(int index, int length) {
    if (index == 0 && heapMemory != null && heapOffset == 0) {
      // Arrays.copyOf is an intrinsics, which is faster
      return Arrays.copyOf(heapMemory, length);
    }
    if (index + length > size) {
      throwIndexOOBExceptionForRead(length);
    }
    byte[] data = new byte[length];
    copyToUnsafe(index, data, Platform.BYTE_ARRAY_OFFSET, length);
    return data;
  }

```